### PR TITLE
Indirect SimulatedDensityOfStates - Refactor file load and general tidy

### LINKED ISSUE #17106: Indirect Simulations SDOS - Refactor
The PyExec for `SimulatedDensityOfStates` is getting rather long and becoming difficult to read. This should be refactored to remove possible duplication and improve readability

### PR DESCRIPTION
**Pull request by ElliotAOram**
Refactor the loading of phonon and castep file from `SimulatedDesnityOfStates` into seperate scripts located in the  `Scripts/Inelastic/DOS` directory. Additionally refactored PyExec to improve readability and reduce number of global variables 

**To test:**
- Code review
- Ensure unit test for `SimulatedDensityOfStates` passes

Fixes #17106 and #13599 

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimulatedDensityOfStates.py [hunk 1]
```diff
@@ -333,11 +249,132 @@ def _get_properties(self):
...
-        self._calc_partial = (len(self._ions) > 0)
+        self._calc_partial = (len(self._ions_of_interest) > 0)
+
+
+#----------------------------------------------------------------------------------------
+
+    def _create_ion_table(self, unit_cell, ions):
+        """
+        Creates an ion table from the data in the phonon/castep file
+        """
```
Comment (KarlPalmen) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimulatedDensityOfStates.py]: I don't see any kind of file in this function.

Comment (KarlPalmen) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimulatedDensityOfStates.py]: I think the phonon/castep data has already been read/loaded into self, in which case the comment should state this. E.g. "... data read from phonon/castep file".

**Comment by KarlPalmen** [comment]
Looks good :shipit: .

**Comment by FedeMPouzols** [comment]
Nice refactoring. I only noticed that in some functions not all the parameters are documented with `@param` (for example in `load_castep.py`) but their meaning is not hard to guess and that can be improved over time.

**Pull request merged**